worker/utils/fastlyJob.js

```diff
@@ -12,6 +12,62 @@ class FastlyJobClass {
     }
   }
 
+  // takes in an array of urls and purges cache for each
+  async purgeCache(urlArray) {
+    if (!Array.isArray(urlArray)) {
+      throw new Error('Parameter `urlArray` needs to be an array of urls');
+    }
+
+    let that = this;
+    let urlCounter = urlArray.length;
+    let purgeMessages = [];
+
+    // the 1 is just "some" value needed for this header: https://docs.fastly.com/en/guides/soft-purges
+    const headers = {
+      'fastly-key': environment.getFastlyToken(),
+      'accept': 'application/json',
+      'Fastly-Soft-Purge': '1',
+    };
+
+    return new Promise((resolve, reject) => {
+      for (let i = 0; i < urlArray.length; i++) {
+        // perform request to purge
+        request({
+          method: 'PURGE',
+          url: urlArray[i],
+          headers: headers,
+        }, function(err, response, body) {
+          // url was not valid to purge
```

Comment:
I thought we can't determine when a url is valid?

Reply:
discussed over DM